### PR TITLE
is_sandboxed: Add support for snaps

### PR DESCRIPTION
**Pull request by A6GibKm**

### CODE REVIEW

File: src/helpers.rs [hunk 1]
```diff
@@ -150,3 +151,68 @@ where
...
+    };
+
+    let mut buffer = String::new();
+    match file.read_to_string(&mut buffer) {
+        Ok(_) => parse_cgroup_v2(&buffer),
+        Err(_) => false,
+    }
+}
+
+fn parse_cgroup_v2(cgroups: &str) -> bool {
```
Comment (A6GibKm) [inline src/helpers.rs:170]: Ideally this returns a Option<None> but it is a bit hard to do so with the current logic, we don't want to return None if the first line is not what we want.

File: src/helpers.rs [hunk 2]
```diff
@@ -150,3 +151,68 @@ where
 pub(crate) fn path_from_null_terminated(bytes: Vec<u8>) -> PathBuf {
     Path::new(OsStr::from_bytes(bytes.split_last().unwrap().1)).to_path_buf()
 }
+
+pub(crate) fn is_snap() -> bool {
```
[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/helpers.rs:155]: Any reason why this one is not async compared to the oo7 one?

[the remark this answers is not in the document]
Comment (A6GibKm) [reply src/helpers.rs:155]: just because doing so requires adding the async_std+tokio dance

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/helpers.rs:155]: That is fine, feel free to add those features

File: src/helpers.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply src/helpers.rs:170]: The method itself is a bit weird, it is called parse_cgroup_v2 but only checks if it is a snap app so. Maybe call it cgroup_v2_is_snap or something like that.

**Pull request merged**